src/qt/guiutil.cpp

```diff
@@ -978,6 +1004,188 @@ QString loadStyleSheet()
     return *stylesheet.get();
 }
 
+QFont::Weight getFontWeightNormal()
+{
+    return fontWeightNormal;
+}
+
+QFont::Weight getFontWeightBold()
+{
+    return fontWeightBold;
+}
+
+int getFontSize()
+{
+    return fontSize;
+}
+
+bool loadFonts()
+{
+    // TODO: Evaluate maybe wrapping this later into some "theme handler" or so..
+    QString family = defaultFontFamily;
+    QString italic = "Italic";
+
+    std::map<QString, bool> mapStyles {
+        {"Thin", true},
+        {"ExtraLight", true},
+        {"Light", true},
+        {"Italic", false},
+        {"Regular", false},
+        {"Medium", true},
+        {"SemiBold", true},
+        {"Bold", true},
+        {"ExtraBold", true},
+        {"Black", true},
+    };
+
+    QFontDatabase database;
+    std::vector<int> vecFontIds;
+
+    for (const auto& it : mapStyles) {
+        QString font = ":fonts/" + family + "-" + it.first;
+        vecFontIds.push_back(QFontDatabase::addApplicationFont(font));
+        qDebug() << __func__ << ": "<< font << " loaded with id " <<  vecFontIds.back();
+        if (it.second) {
+            vecFontIds.push_back(QFontDatabase::addApplicationFont(font + italic));
+            qDebug() << __func__ << ": "<< font + italic << " loaded with id " <<  vecFontIds.back();
+        }
+    }
+
+    // Fail if an added id is -1 which means QFontDatabase::addApplicationFont failed.
+    if (std::find(vecFontIds.begin(), vecFontIds.end(), -1) != vecFontIds.end()) {
+        return InitError("Font loading failed.");
+    }
+
+    // Print debug logs for added fonts fetched by the added ids
+    for (auto i : vecFontIds) {
```

Comment:
Is there a reason why this should be const reference in this case? `vecFontIds` just contains `int` so it doesn't really matter, does it? ;)

Anyway, still adjusted because it also doesn't hurt to have the const reference.